### DISCUSSION TITLE
Build Dicom Worklist Server by Mirth Connect

### DISCUSSION THREAD

**Discussion by hungndqps** [nextgenhealthcare/connect#5210] [Q&A] [answered]
hi guys, I'm a newbie, I have one question for you and I hope this problem could be solved. I want to build a dicom worklist server with open sources so that I found mirth connect and I think it will be usefull because a lot of topics and forum talk about this. Please help me. Thank you so much.

**Comment by marcheschi** [comment, marked as answer]
Hi we use dcm4chee dicom server https://dcm4che.atlassian.net/wiki/spaces/ee2/overview
the version 2.x is simpler  and it is used as worklist server in big hospitals.
You can create orm messages with mirth and send them to the worklist server.
Best
Paolo

**Reply by hungndqps** [reply]
Thanks for your help

**Reply by luistelmocosta** [reply]
> Hi we use dcm4chee dicom server https://dcm4che.atlassian.net/wiki/spaces/ee2/overview the version 2.x is simpler and it is used as worklist server in big hospitals. You can create orm messages with mirth and send them to the worklist server. Best Paolo

Hello! Can you please give me some more information about this? I am working with a dcm4chee dicom server and I was wondering how I could use it as a worklist server. Do you have any resources about this that could share?

Regards, 
Luís

**Reply by hungndqps** [reply]
> > Hi we use dcm4chee dicom server https://dcm4che.atlassian.net/wiki/spaces/ee2/overview the version 2.x is simpler and it is used as worklist server in big hospitals. You can create orm messages with mirth and send them to the worklist server. Best Paolo
> 
> Hello! Can you please give me some more information about this? I am working with a dcm4chee dicom server and I was wondering how I could use it as a worklist server. Do you have any resources about this that could share?
> 
> Regards, Luís

hello Luis, you used dcm4chee light version, right ? can you share me some ways to deploy it. Like system requirements, capacity of hard disk or somethings ?

Best Regards, 
Hungndq

**Reply by luistelmocosta** [reply]
Hey @hungndqps. The system was provided to me, it wasn't me that implemented it. Sorry but I don't think I can help you with that.

**Reply by marcheschi** [reply]
Hi
If you use dcm4chee as a worklist server the system requirements are very low : 
We use a VM with 20 GB of HD and 4 GB of ram 
The server is working out of the box you can only have to send ORM messages to it.
Best Regards
Paolo

**Reply by luistelmocosta** [reply]
What if I want to query the dcm4chee worklist server and forward the DICOM image to another service? Is that possible?

Regards, 
Luís

**Reply by marcheschi** [reply]
The worklist server is used to to serve the Modality Worklist to the modality.
What you want to do is possible but you have to use the archive feature of dcm4chee not the worklist.
Paolo

**Reply by luistelmocosta** [reply]
I believe you are talking about the WADO (Web Access to DICOM Objects) feature, right?

**Reply by marcheschi** [reply]
Hi
You can use wado to retrieve DICOM files from the server.
Paolo

**Reply by luistelmocosta** [reply]
Thank you for the replies, I get a better overview of the problem now. 
One last thing, what kind of source connector would suit my needs? Should it be done by using the HTTP Request? Or using something like the `dcmqr` tool? 

Best,
Luís

**Reply by marcheschi** [reply]
There are two version of DCM4CHEE Archive, the old one dcm4chee-2.x and the new one dcm4chee-arc-light
For the first I use dcm4che library to move dicom files
for the second you can use also  QIDO-RS (Query based on ID for DICOM Objects by RESTful Services) (with http requests)
Paolo

**Reply by luistelmocosta** [reply]
The new one (dcm4chee-arc-light) was the one provided to me and I was trying with QIDO-RS, indeed. But from what I could understand QIDO-RS is a Query based on ID but If I want to move the DICOM object I need to use WADO-RS, right? How would that work with QIDO-RS?

Thank you very much, I think I'm getting close :)

Best, 
Luís

**Reply by marcheschi** [reply]
You can use Quido-rs to make queries and wado-rs to retrieve instances or studies
For example to get a study:
/aets/{aet}/rs/studies/{study}
curl -X 'GET' \
  'http://testserver:8080/dcm4chee-arc/aets/ed/rs/studies/123?dicomdir=false' \
  -H 'accept: multipart/related;type=application/dicom'
Something like that.
Refer to the standards for further examples.
Paolo

**Reply by luistelmocosta** [reply]
Yes, I got that part correctly, thank you. 
My main doubt is how to copy a specific DICOM Object using the ID I got from QIDO-RS.
I understand that I can use an HTTP Request to get the ID from the QIDO feature but then which kind of connector would get the DICOM object so I can send it to another service (in my case it is the Azure Health Service)?

Best, 
Luís

**Reply by marcheschi** [reply]
In this case is simpler You can invoke the sending directly from dcm4chee-arc to Azure using classic cmove without transferring the study trough mirth.
Paolo

**Reply by luistelmocosta** [reply]
Do you have any documentation/examples that I can read about the solution you suggested? 

Best, 
Luís

**Reply by hungndqps** [reply]
> Hi we use dcm4chee dicom server https://dcm4che.atlassian.net/wiki/spaces/ee2/overview the version 2.x is simpler and it is used as worklist server in big hospitals. You can create orm messages with mirth and send them to the worklist server. Best Paolo

hello, I got a problem. After I deployed MWL Server (include Mirth and DCM4CHEE arc). I have sent Hl7 Order Messages to Modalities. By the way, DCM4CHEE Arc supported OMG O19 message to cancel worklist. Do you have any message templates about this ? Thanks.
![image](https://user-images.githubusercontent.com/72735306/184057683-e622725f-318c-4582-a565-96251701e0ca.png)

**Comment by mjhca** [comment]
There is no MWL listener capability currently in Mirth as of v4.01.  

The "DICOM listener" is only a "DICOM Storage Listener" (C-STORE) which wraps the dcmrcv code from dcm4che2.

I provided a patch in 2016 (see #1149) to support various SCPs (C-STORE, C-FIND, etc) with working code to act as a MWL server.  But it appears the attachment was lost in the conversion of the issues system to Github, and the issue was closed due to inactivity.  

@narupley @rbeckman-nextgen , would it be possible to restore the attachment on the issue and possibly re-open it?  This would continue to be a helpful feature if it could get incorporated into the codebase someday.

Original JIRA link with patch attachment was here but is no longer accessible https://www.mirthcorp.com/community/issues/browse/MIRTH-1147

**Reply by mjhca** [reply]
Example of valid XML destination response in #1149 was also mangled when imported to Github from Jira.  Should read:

```
<dicomItems>
<dicom><tag00100010 len="4" tag="00100010" vr="PN">PatientName1</tag00100010></dicom>
<dicom><tag00100010 len="4" tag="00100010" vr="PN">PatientName2</tag00100010></dicom>
</dicomItems>
```
